lib/Optimizer/GraphOptimizer.cpp

```diff
@@ -855,6 +856,81 @@ static void optimizeBatchNorm(Function *F) {
   } // For all nodes in the graph.
 }
 
+/// Check if the input of this method (vector of NodeValues) can be
```

Comment:
This function does more than seeing if a series of Nodes can be concatenated -- e.g. the `origCN` would not be necessary if that was the case. Also, what happens if there is more than one axis that the concat can take place? Please add more details here.

Reply:
`what happens if there is more than one axis that the concat can take place?`
The only case (IMHO) I can figure out for this statement, is that all the inputs share the same dims vector.  For this case, i go over the dimensions from right to left, find the first index that can make the cumulative number of units divided by the minimal number of units to legally concatenate. Because we want the small tensors that got concatenated keep their inside data order.